src/wp-admin/includes/plugin.php

```diff
@@ -1414,6 +1414,37 @@ function add_comments_page( $page_title, $menu_title, $capability, $menu_slug, $
 	return add_submenu_page( 'edit-comments.php', $page_title, $menu_title, $capability, $menu_slug, $function );
 }
 
+/**
+ * Add submenu page to the Security main menu.
+ *
+ * The function which is hooked in to handle the output of the page must check
+ * that the user has at least 'manage_options' capability.
+ *
+ * @param string   $page_title The text to be displayed in the title tags of the page when the menu is selected.
+ * @param string   $menu_title The text to be used for the menu.
+ * @param string   $menu_slug  The slug name to refer to this menu by; must match an active plugin slug.
+ * @param callable $function   The function to be called to output the content for this page.
+ * @return false|string The resulting page's hook_suffix, or false if the user does not have the 'manage_options' capability.
+ */
+function add_security_page( $page_title, $menu_title, $menu_slug, $function = '' ) {
+    foreach ( get_option( 'active_plugins' ) as $path ) {
+        $parts = explode( '/', $path );
+        if ( $menu_slug === $parts[0] ) {
+            add_filter(
+                'plugin_action_links_' . $path,
+                function ( $links ) use ( $menu_slug ) {
+                    array_unshift( $links, sprintf( '<a href="%s?page=%s" title="%s"><span class="dashicon dashicons-shield"></span></a>', admin_url( 'security.php' ), $menu_slug, __( 'Security' ) ) );
+                    return $links;
+                },
```

Comment:
Filters in public code should always use named functions so they can be unhooked later if needed.

Reply:
The only way I can see to do that is by adding yet another global. In this case we're only adding an item to an array, so if I set the priority to `PHP_INT_MAX-1` people can still pull it out.
Your call - another global or change the priority.